src/main/java/com/thealgorithms/sorts/TopologicalSort.java

```diff
@@ -0,0 +1,170 @@
+package com.thealgorithms.sorts;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+
+/**
+ * The Topological Sorting algorithm linearly orders a DAG or Directed Acyclic Graph into
+ * a linked list. A Directed Graph is proven to be acyclic when a DFS or Depth First Search is
+ * performed, yielding no back-edges.
+ *
+ * https://en.wikipedia.org/wiki/Topological_sorting
+ *
+ * @author Jonathan Taylor (https://github.com/Jtmonument)
+ * Based on Introduction to Algorithms 3rd Edition
+ */
+public class TopologicalSort {
+
+     /*
+     * The linked list that will contain the vertices in linear order.
+     * */
+    private static final LinkedList<Vertex> list = new LinkedList<>();
+
+     /*
+     * Enum to represent the colors for the depth first search
+     * */
+    private enum Color {
+        WHITE, GRAY, BLACK
+    }
+
+     /*
+     * Class to represent vertices
+     * */
+    static class Vertex {
+        public final String label;  // name of vertex
+        public int weight;  // weight of vertex (more accurately defined as the start time in DFS)
+        public int finished;    // time that the vertex has finished a visit in DFS
+        public Vertex predecessor;  // parent of the vertex
+        public Color color = Color.WHITE;   // Represents the category of visit in DFS
+        public final ArrayList<String> next = new ArrayList<>();    // array of names of descendant vertices
+
+        public Vertex(String label) {
+            this.label = label;
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder edges = new StringBuilder();
+            edges.append(predecessor == null ? "NONE" : predecessor.label).append("\n\tEdges:");
+            next.forEach(vertex -> edges.append("\n\t\t").append(label).append(" -> ").append(vertex));
+            return "Vertex " + label +
+                    ":\n\tWeight: " + weight +
+                    "\n\tFinished: " + finished +
+                    "\n\tColor: " + color +
+                    "\n\tPredecessor: " +
+                    edges;
+        }
+
+    }
+
+     /*
+     * Graph class uses the adjacency list representation
+     * */
+    static class Graph {
+
+        private final HashMap<String, Vertex> adj = new HashMap<>();    // adjacency list representation
+        private int time;   //time variable in DFS
+        private boolean isAcyclic;  // Checks if the graph is acyclic
+
+        /*
+        * Print the sorted linear order of vertices
+        * */
+        public void print() {
+            list.forEach(System.out::println);
+            if (isAcyclic)
+                System.err.println("This graph contains a cycle. No linear ordering is possible.");
```

Comment:
Acyclic graphs do not contain cycles by definition

Reply:
The code works as intended. That's a silly semantic error on my part. Line 128 will be changed to false and line 76 will be the result of the not operator